src/python/pants/backend/python/target_types_rules.py

```diff
@@ -135,29 +142,124 @@ async def inject_pex_binary_entry_point_dependency(
 # -----------------------------------------------------------------------------------------------
 
 
+@rule(desc="Determining the entry points for a `python_distribution` target")
+async def resolve_python_distribution_entry_points(
+    request: ResolvePythonDistributionEntryPointsRequest,
+) -> ResolvedPythonDistributionEntryPoints:
+    field_value = request.entry_points_field.value
+    if field_value is None:
+        return ResolvedPythonDistributionEntryPoints(None)
+
+    address = request.entry_points_field.address
+    entry_points: dict[str, dict[str, EntryPoint]] = defaultdict(dict)
+
+    for section, values in field_value.items():
+        for name, ref in values.items():
+            if ref.startswith(":") or "/" in ref:
+                targets = await Get(Targets, UnparsedAddressInputs([ref], owning_address=address))
+                if not targets:
+                    raise InvalidFieldException(
+                        f'The "entry_points" target address {repr(ref)} does not resolve to any known target.'
+                    )
+
+                for target in targets:
```

Comment:
There are no globs for addresses, so you can assume `targets` is exactly 1. Fwit, you can add an `AssertionError` if that is not the case.

This will allow you to remove the `for` loop nesting.